src/exec.js

```diff
@@ -58,19 +60,16 @@ function execSync(cmd, opts) {
   if (fs.existsSync(codeFile)) common.unlinkSync(codeFile);
 
   var execCommand = '"'+process.execPath+'" '+scriptFile;
-  var execOptions = {
-    env: process.env,
-    cwd: _pwd(),
-    maxBuffer: 20*1024*1024
-  };
-
+  opts.cwd = opts.cwd || _pwd();
+  opts.env = opts.env || process.env;
+  opts.maxBuffer = opts.maxBuffer || DEFAULT_MAXBUFFER_SIZE;
```

Comment:
@nfischer: But you could do this, and it would have the same effect (I think):

```javascript
opts = common.extend({
  silent: common.config.silent,
  cwd: _pwd(),
  env: process.env,
  maxBuffer: DEFAULT_MAXBUFFER_SIZE,
}, opts);
```